### ISSUE TITLE
Stopping Criteria is Not Working Correctly

### ISSUE THREAD

**Issue by btcross26** [btcross26/genestboost#18]
Issue: Stopping Criteria is not working correctly on the training set or validation set.

Cause: The training set criteria is using absolute value for the top parameter. This allows training set error to increase as long as in the increase is below `tol`, which is not the intent. The validation set stopping criteria needs a second index to identify the second column of the internal training loss storage - otherwise, a list is compared to a list and the stopping criteria does not work correctly - this became obvious when models where not stopping despite obvious increase in holdout set error.
